Sources/Amplitude/Amplitude.m

```diff
@@ -284,6 +272,49 @@ - (instancetype)initWithInstanceName:(NSString *)instanceName {
     return self;
 }
 
+- (void)migrateToFileStorage {
+     NSString *databaseDirectory = [AMPUtils platformDataDirectory];
+     NSString *databasePath = [databaseDirectory stringByAppendingPathComponent:@"com.amplitude.database"];
+    if (![self->_instanceName isEqualToString:kAMPDefaultInstance]) {
+        databasePath = [NSString stringWithFormat:@"%@_%@", databasePath, self.instanceName];
+    }
+    NSString *fileStoragePath =  [AMPStorage getAppStorageAmpDir:self.instanceName];
+
+    BOOL firstMigration = ![[NSFileManager defaultManager] fileExistsAtPath:fileStoragePath isDirectory: true];
+    BOOL needDBMigrate =  ([[NSFileManager defaultManager] fileExistsAtPath:databasePath] && firstMigration) ? true : false;
+    if (!needDBMigrate) return;
+    int eventCount = [self.dbHelper getEventCount];
+    if (eventCount > 0) {
+        NSArray *events = [self.dbHelper getEvents: -1 limit: -1];
+        NSDictionary *eventsDict = [self mergeEventsAndIdentifys:events identifys:nil numEvents:eventCount];
+        long long maxEventId = [[eventsDict objectForKey:MAX_EVENT_ID] longLongValue];
+        for(NSString *event in events) {
+            [AMPStorage storeEvent:event instanceName:self.instanceName];
+        }
+       [self.dbHelper removeEvents:maxEventId];
+    }
+
+    int identifyCount = [self.dbHelper getIdentifyCount];
+    if (identifyCount > 0) {
+        NSArray *identifys = [self.dbHelper getIdentifys: -1 limit: -1];
+        NSDictionary *identifysDict = [self mergeEventsAndIdentifys:nil identifys:identifys numEvents:identifyCount];
+        long long maxIdentifyId = [[identifysDict objectForKey:MAX_IDENTIFY_ID] longLongValue];
+        for(NSString *identify in identifys) {
+            [AMPStorage storeIdentify:identify instanceName:self.instanceName];
+        }
+        [self.dbHelper removeIdentifys:maxIdentifyId];
+    }
+
+    // migrate for store database table
+    [self->_defaultDataStorage setObject:[self.dbHelper getValue:USER_ID] forKey:USER_ID];
```

Comment:
This is very dangerous since our user is likely to use NSUserDefaults too. They might have the same key for USER_ID and other keys below.

Reply:
Need to add prefix for the key. 
e.g. amplitude.user_id OR amplitude_user_id

Reply:
better update the constant and create constants for OLD_USER_ID

Reply:
Also do we need to consider instanceName here?